cmake/create_test.cmake

```diff
@@ -101,63 +101,31 @@ function(ginkgo_create_cuda_test test_name)
     add_test(NAME ${REL_BINARY_DIR}/${test_name} COMMAND ${TEST_TARGET_NAME})
 endfunction(ginkgo_create_cuda_test)
 
-function(ginkgo_create_hip_test_special_linkage test_name)
```

Comment:
Do you know why special link works?

Reply:
What I understand is that the special linkage was required because it added the libraries of `HIP`, like `libamdhip64.so` in an automatic fashion. What happens now is that there is a public dependency to that library directly (the `.so` and not a target), which means that this dependency is always propagated including to our tests. Thus, the tests can just link like normal, they will anyway also link against the HIP libraries through that, which makes it all work in the default case.

See: https://github.com/ginkgo-project/ginkgo/blob/fix_hip/hip/CMakeLists.txt#L264

Reply:
I see. I also put the previous comments link here for reference.
https://github.com/ginkgo-project/ginkgo/pull/393#issuecomment-558626356